components/com_config/view/modules/tmpl/default.php

```diff
@@ -15,7 +15,22 @@
 JHtml::_('behavior.combobox');
 JHtml::_('formbehavior.chosen', 'select');
 
-$hasContent = empty($this->item['module']) || $this->item['module'] === 'custom' || $this->item['module'] === 'mod_custom';
+jimport('joomla.filesystem.file');
+
+$editorText  = false;
+$moduleXml   = JPATH_SITE . '/modules/' . $this->item['module'] . '/' . $this->item['module'] . '.xml';
+
+if (JFile::exists($moduleXml))
+{
+	$xml = simplexml_load_file($moduleXml);
+
+	if (isset($xml->customContent))
+	{
+		$editorText = true;
+	}
+}
+
+$hasContent = empty($this->item['module']) || $this->item['module'] === 'custom' || $this->item['module'] === 'mod_custom' || $editorText === true;
```

Comment:
I think the complete condition is wrong, I would suggest:

```suggestion
$hasContent = (!empty($this->item['module']) && ($this->item['module'] === 'custom' || $this->item['module'] === 'mod_custom')) || $editorText === true;
```

if this is the original query in the backend it should be changed there too.

Reply:
TBH. I would not know which query is right or wrong.

Reply:
@HLeithner 
Just checked this. I think your suggestion makes sense, including for the similar code in backend edit.php

But I I found the same code in 1.6.0...
What shall we do? Is there a risk to not be B/C ?

Also, I find a similar code in 1.5.26...
```
		<?php
		if ( !$row->module || $row->module == 'custom' || $row->module == 'mod_custom' ) {
			?>
			<fieldset class="adminform">
				<legend><?php echo JText::_( 'Custom Output' ); ?></legend>

				<?php
				// parameters : areaname, content, width, height, cols, rows
				echo $editor->display( 'content', $row->content, '100%', '400', '60', '20', array('pagebreak', 'readmore') ) ;
				?>

			</fieldset>
			<?php
		}
		?>
```
🤒

Reply:
Is it possible that $this->module is empty? 
Shouldn't be or your PR doesn't check on empty some lines above.

```$moduleXml   = JPATH_SITE . '/modules/' . $this->item['module'] . '/' . $this->item['module'] . '.xml';```

Reply:
It checks correctly the xml
```

$moduleXml   = JPATH_SITE . '/modules/' . $this->item['module'] . '/' . $this->item['module'] . '.xml';
if (JFile::exists($moduleXml))
{
	$xml = simplexml_load_file($moduleXml);
	if (isset($xml->customContent))
	{
		$editorText = true;
	}
}
```

I honestly do not understand either why we have for ages in backend
`$hasContent = empty($this->item->module) ||  isset($this->item->xml->customContent);`
on the model of 1.5 using row